{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/nice-digital/text-classifier/blob/main/text-classifier-lr.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **Text classifier Colab**\n",
        "\n",
        "This Colab notebook allows you to categorise a set of scientific papers into two categories. This is experimental code\n",
        "\n",
        "**Note**: Name your training file *training.csv*  and test file *testing.csv* (*title* column should be named 'Title' or 'title' and *abstract* column if present should be named 'Abstract' or 'abstract'), and upload it by pressing the upload button on the top left of the left sidebar. The results will appear in a folder named *RESULTS*. RESULTS folder will be automatically created by the code.\n"
      ],
      "metadata": {
        "id": "J3XQyHVQAmMV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Install Python packages { form-width: \"20%\" }\n",
        "\n",
        "#@markdown Please execute this cell by pressing the _Play_ button\n",
        "#@markdown on the left to download and import third-party software\n",
        "#@markdown in this Colab notebook.\n",
        "\n",
        "#@markdown This installs the software on the Colab\n",
        "#@markdown notebook in the cloud and not on your computer.\n",
        "from IPython.utils import io\n",
        "try:\n",
        "  with io.capture_output() as captured:\n",
        "    %shell pip install scispacy\n",
        "    # %shell pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.0/en_core_sci_md-0.5.0.tar.gz\n",
        "    # %shell pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.1/en_ner_bc5cdr_md-0.5.1.tar.gz\n",
        "    # %shell pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.1/en_ner_bionlp13cg_md-0.5.1.tar.gz\n",
        "    %shell pip install pyLDAvis==2.1.2\n",
        "    %shell pip install import-ipynb\n",
        "    %shell pip install pandas\n",
        "    %shell pip install shutup\n",
        "\n",
        "except subprocess.CalledProcessError:\n",
        "  print(captured)\n",
        "  raise\n",
        "import shutup\n",
        "shutup.please()\n",
        "\n",
        "import os\n",
        "import numpy as np\n",
        "import spacy\n",
        "import scispacy\n",
        "import pandas as pd\n",
        "from scispacy.abbreviation import AbbreviationDetector\n",
        "\n",
        "from pathlib import Path\n",
        "import collections\n",
        "import csv\n",
        "import multiprocessing as mp\n",
        "from multiprocessing import Pool\n",
        "\n",
        "from sklearn.metrics import f1_score\n",
        "from sklearn.metrics import accuracy_score\n",
        "from sklearn.metrics import precision_score\n",
        "from sklearn.metrics import recall_score\n",
        "from sklearn.metrics import f1_score\n",
        "from sklearn.metrics import roc_auc_score\n",
        "from sklearn.metrics import confusion_matrix\n",
        "from sklearn.metrics import ConfusionMatrixDisplay\n",
        "\n",
        "from sklearn.feature_extraction.text import TfidfVectorizer\n",
        "from sklearn.linear_model import LogisticRegression\n",
        "\n",
        "from sklearn.pipeline import Pipeline\n",
        "from sklearn.model_selection import train_test_split\n",
        "from fastai.imports import *\n",
        "\n",
        "\n",
        "cpu_count = mp.cpu_count()\n",
        "\n",
        "pd. set_option('display.max_colwidth', None)"
      ],
      "metadata": {
        "id": "yNF5tewRA4Dn",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Create train/test datasets from human/animal datasets { form-width: \"20%\" }\n",
        "animal = pd.read_csv('excludes_Animal_2200.csv')\n",
        "human = pd.read_csv('includes_human_2400.csv')\n",
        "\n",
        "#add target variable\n",
        "animal['target'] = 0\n",
        "human['target'] = 1\n",
        "\n",
        "print(animal.columns)\n",
        "print(human.columns)\n",
        "\n",
        "#combine & shuffle the datasets\n",
        "combined_data = pd.concat([animal, human], axis=0)\n",
        "shuffled_combined_df = combined_data.sample(frac=1).reset_index(drop=True)\n",
        "\n",
        "#create a 80-20 split from it\n",
        "training, testing = train_test_split(shuffled_combined_df, test_size=0.2, random_state=42)\n"
      ],
      "metadata": {
        "id": "E46oKVfi0KHs",
        "outputId": "b5db9533-813e-4b93-a90a-05760ece206d",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Index(['Title', 'Abstract', 'Primary Author', 'Journal', 'Year', 'Volume',\n",
            "       'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'],\n",
            "      dtype='object')\n",
            "Index(['Title', 'Abstract', 'Primary Author', 'Journal', 'Year', 'Volume',\n",
            "       'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'],\n",
            "      dtype='object')\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title File settings to get started  { form-width: \"20%\" }\n",
        "\n",
        "#@markdown Please ensure the training.csv and testing.csv are uploaded and execute this cell by pressing the _Play_ button\n",
        "#@markdown on the left\n",
        "\n",
        "#@markdown The training.csv and testing.csv files should have 'title', optional 'abstract' fields. Additionally the file should have a 'target' field\n",
        "#@markdown which indicates whether the title/abstract is an include (coded as 1) or exclude (coded as 0)\n",
        "TRAIN_PATH = 'training.csv'\n",
        "TEST_PATH = 'testing.csv'\n",
        "\n",
        "results_folder = 'RESULTS'\n",
        "RESULTS_FOLDER = results_folder     #***user input\n",
        "if not os.path.isdir(RESULTS_FOLDER):\n",
        "    os.makedirs(RESULTS_FOLDER)\n",
        "RESULTS_PATH = Path(RESULTS_FOLDER)"
      ],
      "metadata": {
        "id": "XDSMaZGDI-Ko"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Read in input data as separate training.csv and testing.csv. **Ignore** this block if human/animal data was uploaded above { form-width: \"20%\" }\n",
        "try:\n",
        "    training = pd.read_csv(TRAIN_PATH)\n",
        "    orig_colnames = training.columns\n",
        "    print(orig_colnames)\n",
        "\n",
        "    testing = pd.read_csv(TEST_PATH)\n",
        "\n",
        "except Exception as e:\n",
        "    print(e)\n",
        "    raise"
      ],
      "metadata": {
        "cellView": "form",
        "id": "49646FC321sm"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Read in input data { form-width: \"20%\" }\n",
        "rename_map = {'Title': 'title', 'Abstract': 'abstract'}\n",
        "training.rename(columns = rename_map, inplace = True)\n",
        "testing.rename(columns = rename_map, inplace = True)\n",
        "print(\"Number of studies in the training dataset: \" + str(training.shape[0]))\n",
        "print(\"Number of studies in the training dataset: \" + str(testing.shape[0]))\n",
        "\n",
        "#rename the columns so that the relevant column names are 'title' and 'abstract'\n",
        "\n",
        "try:\n",
        "  training['title_orig'] = training['title']\n",
        "  testing['title_orig'] = testing['title']\n",
        "except Exception as e:\n",
        "  print(e)\n",
        "  print(\"Error- No title detected! Title is needed!\")\n",
        "  raise\n",
        "\n",
        "# drop any duplicates based on 'title'\n",
        "training.drop_duplicates(subset=['title'], inplace=True)\n",
        "testing.drop_duplicates(subset=['title'], inplace=True)\n",
        "print(\"Number of studies in the training dataset after de-dupe: \" + str(training.shape[0]))\n",
        "print(\"Number of studies in the testing dataset after de-dupe: \" + str(testing.shape[0]))\n",
        "\n",
        "training['titleabstract'] = training['title'] + \" \" + training['abstract']\n",
        "training['titleabstract'] = training['titleabstract'].str.lower()\n",
        "\n",
        "testing['titleabstract'] = testing['title'] + \" \" + testing['abstract']\n",
        "testing['titleabstract'] = testing['titleabstract'].str.lower()"
      ],
      "metadata": {
        "id": "eR8-4xBMB7nQ",
        "outputId": "64f1e726-4d4d-47da-873b-c0ebb0a6b8d4",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Number of studies in the training dataset: 3693\n",
            "Number of studies in the training dataset: 925\n",
            "Number of studies in the training dataset after de-dupe: 3693\n",
            "Number of studies in the testing dataset after de-dupe: 925\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Fit logistic regression model (in progress) { form-width: \"20%\" }\n",
        "\n",
        "#A sklearn pipeline comprising of tf-idf vectorizer (using tri-gram) and logistic regression model. The parameters for logistic regression\n",
        "#are taken from prior hyper-parameter tuning.\n",
        "text_clf = Pipeline([\n",
        "                ('tfidfvect', TfidfVectorizer(ngram_range = (3,3), stop_words = 'english')),\n",
        "                ('clf', LogisticRegression(C=100, max_iter = 5000, solver = 'liblinear', penalty = 'l2', class_weight = 'balanced')),\n",
        "               ])\n",
        "y_train = training['target']\n",
        "model = text_clf.fit(training['titleabstract'].astype(str),y_train)\n",
        "\n"
      ],
      "metadata": {
        "id": "0ivxu5aJBbR5"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Predict category and evaluate performance (in progress) { form-width: \"20%\" }\n",
        "\n",
        "#Using the model that was fit to the training data above, evaluate the model's performance on test data.\n",
        "data = testing['titleabstract'].astype(str)\n",
        "y_test = testing['target']\n",
        "yhat = model.predict(data)\n",
        "yhat_probs = model.predict_proba(data)[:,1]\n",
        "yhat_adjusted = np.zeros(data.shape[0], dtype=int)\n",
        "THRESHOLD = 0.4\n",
        "yhat_adjusted[yhat_probs >= THRESHOLD] = 1\n",
        "\n",
        "report_dict = {}\n",
        "decimal_places = 3\n",
        "report_dict['Accuracy'] = accuracy_score(y_test, yhat_adjusted).round(decimal_places)\n",
        "report_dict['Precision'] = precision_score(y_test,yhat_adjusted).round(decimal_places)\n",
        "report_dict['Recall'] = recall_score(y_test, yhat_adjusted, average = 'binary').round(decimal_places)\n",
        "report_dict['F1-Score'] = f1_score(y_test, yhat_adjusted).round(decimal_places)\n",
        "report_dict['ROC_AUC'] = roc_auc_score(y_test, yhat_adjusted).round(decimal_places)\n",
        "cm = confusion_matrix(y_test, yhat_adjusted)\n",
        "FP = cm[0][1]\n",
        "TN = cm[0][0]\n",
        "FN = cm[1][0]\n",
        "TP = cm[1][1]\n",
        "specificity = (TN / (TN+FP)).round(decimal_places)\n",
        "FPR = (FP/(FP+TN)).round(decimal_places)\n",
        "FNR = (FN/(FN+TP)).round(decimal_places)\n",
        "report_dict['FPR'] = FPR\n",
        "report_dict['FNR'] = FNR\n",
        "report_dict['Specificity'] = specificity\n",
        "\n",
        "print('Classification report:\\n{}'.format(report_dict))\n"
      ],
      "metadata": {
        "id": "hdVqVB-BIfVk",
        "cellView": "form",
        "outputId": "d95029f8-6350-42e4-b0e9-f9314314d106",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Classification report:\n",
            "{'Accuracy': 0.826, 'Precision': 0.764, 'Recall': 0.974, 'F1-Score': 0.856, 'ROC_AUC': 0.816, 'FPR': 0.343, 'FNR': 0.026, 'Specificity': 0.657}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Data Preprocessing\n",
        "Convert text data to numerical features using TF-IDF"
      ],
      "metadata": {
        "id": "RFhHJYnhafa8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "animal = pd.read_csv('excludes_Animal_2200.csv')\n",
        "human = pd.read_csv('includes_human_2400.csv')\n",
        "\n",
        "#animal.head(), human.head()"
      ],
      "metadata": {
        "id": "DsMgFP-AmvTd"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# prompt: shape of the df\n",
        "(animal.shape), (human.shape)"
      ],
      "metadata": {
        "id": "1E7FaSbdbWYq",
        "outputId": "b896d18d-69be-427f-dbc4-21ccf922913c",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "((2212, 10), (2411, 10))"
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#add target variable\n",
        "animal['target'] = 0\n",
        "human['target'] = 1"
      ],
      "metadata": {
        "id": "3VNfccMugnOz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "animal.head()"
      ],
      "metadata": {
        "id": "M6ne7VOxnm02",
        "outputId": "176a3bd9-1980-46a7-cea8-e9dc7a35a2de",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                                                                                                                Title  \\\n",
              "0                                            An In Vivo Comparison: Novel Mesh Suture Versus Traditional Suture-Based Repair in a Rabbit Tendon Model   \n",
              "1  Effects of urea supplementation on ruminal fermentation characteristics, nutrient intake, digestibility, and performance in sheep: A meta-analysis   \n",
              "2                                                 The in vitro and in vivo anti-inflammatory activities of triterpene saponins from Clematis florida.   \n",
              "3                                                             Evaluating Mouse Fibroblast Interaction with Implant Surfaces in a 3D Microenvironment.   \n",
              "4              Deep learning-based image-analysis algorithm for classification and quantification of multiple histopathological lesions in rat liver.   \n",
              "\n",
              "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Abstract  \\\n",
              "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Purpose: Despite advancements in surgical techniques, suture pull-though and rupture continue to limit the early range of motion and functional rehabilitation after flexor tendon repairs. The aim of this study was to evaluate a suturable mesh compared with a commonly used braided suture in an in vivo rabbit intrasynovial tendon model. Method(s): Twenty-four New Zealand female rabbits (3-4 kg) were injected with 2 units/kg botulinum toxin evenly distributed into 4 sites in the left calf. After 1 week, the animals underwent surgical tenotomy of the flexor digitorum tendon and were randomized to repair with either 2-0 Duramesh suturable mesh or to 2-0 Fiberwire using a 2-strand modified Kessler and 6-0 polypropylene running epitendinous suture. Rabbits were killed at 2, 4, and 9 weeks after surgery. Result(s): Grouping across time points, 58.3% (7 of 12) of Duramesh repairs were found to be intact for the explant compared with 16.7% (2 of 12) of Fiberwire repairs (P = .09). At 2 weeks, the mean Duramesh repairs were significantly stronger than the Fiberwire repairs with a mean failure load of 50.7 +/- 12.7 N compared to 14.8 +/- 18.3 N (P = .02). The load supported by the Duramesh repairs at 2 weeks (mean 50.7 +/- 12.7 N) was similar to the load supported by both Fiberwire (52.2 +/- 13.6 N) and Duramesh (57.6 +/- 22.3 N) at 4 weeks. The strength of repair between Fiberwire and Duramesh at 4 weeks and 9 weeks was not significantly different. Conclusion(s): The 2-strand tendon repair with suturable mesh achieved significantly greater strength at 2 weeks than the conventional suture material. Future studies should evaluate the strength of repair prior to 2 weeks to determine the strength curve for this novel suture material. Clinical Relevance: This study evaluates the utility of a novel suturable mesh for flexor tendon repair in an in vivo rabbit model compared with conventional suture material.Copyright ? 2021 The Authors   \n",
              "1  Background and Aim: As a non-protein nitrogen source, urea is a popular, low cost, and easily obtained protein supplement. The objective of the present study was to perform a meta-analysis of the effects of urea supplementation on rumen fermentation and sheep performance. Material(s) and Method(s): A total of 32 experiments from 21 articles were compiled into a dataset. The levels of dietary urea varied from 0 to 31 g/kg of dry matter (DM). Parameters observed were rumen fermentation product, nutrient intake, nutrient digestibility, and sheep performance. This dataset was analyzed using a mixed model methodology, with urea supplementation levels as fixed effects and the different experiments as random effects. Result(s): Increasing levels of urea were associated with increases (p=0.008) in rumen pH, butyrate (C4) production, and ammonia (NH3-N) concentration. Urea supplementation had minor effects on total volatile fatty acids (p=0.242), total protozoa (p=0.429), and the microbial N supply (p=0.619), but tended to increase methane production (CH4; p<0.001). Supplementation of urea increased the intake of dry matter (DM; p=0.004) and crude protein (CP; p=0.001). Digestibility parameters, such as DM digestibility (DMD) and CP digestibility (CPD), also increased (p<0.01) as a result of urea supplementation. Retained N (p=0.042) and N intake (p<0.001) were higher with increasing levels of urea supplementation. In terms of animal performance, supplementation of urea increased average daily gain (ADG; p=0.024), but decreased the hot carcass weight percentage (p=0.017). Conclusion(s): This meta-analysis reports the positive effects of urea supplementation on rumen fermentation products (i.e., pH, C4, and NH3-N), intake (DM, CP, and N), digestibility (DMD and CPD), and ADG in sheep.Copyright: Wahyono, et al. Open Access. This article is distributed under the terms of the Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/), which permits unrestricted use, distribution, and reproduction in any medium, provided you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons license, and indicate if changes were made. The Creative Commons Public Domain Dedication waiver (http://creativecommons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated.   \n",
              "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Clematis florida is widely used in She Ethnopharmacy in China owing to its significant anti-inflammatory activities. This study aimed to investigate the anti-inflammatory effect of the active fraction of C. florida (CFAF) in an arthritis animal model and its possible mechanism. Pre-inflammatory cytokine levels were examined by ELISA. CFAF can significantly improve the symptoms of arthritis such as paw swelling, arthritic index, and histological condition in AA rat. CFAF can also reduce levels of IL-1?, TNF-? and IL-6. Further studies showed that triterpene saponins from CFAF induced anti-inflammatory activity inhibited inflammatory mediators by blocking JAK/STAT signalling pathways in the LPS-treated macrophages.   \n",
              "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Purpose: Previous studies assessing fibroblast interactions with implants have mainly relied on measurements such as cell migration, gene expression, and cell adhesion. For these studies, testing cellular behavior at the implant surface was done by imaging the cell-implant interface using standard microscopy techniques in 2D tissue culture dishes. The true behavior of cells relative to the implant can best be assessed in a more physiologic 3D microenvironment. Materials and Methods: The embedding of the implant disks in 3D collagen gels was standardized with labeled fibroblasts to allow the imaging of fibroblast morphology and behavior when proximal to or binding to the implant disks. This allowed comparison of the behavior of laser-microgrooved and machined implant disk surfaces quantitatively in an in vitro 3D microenvironment. Results: This in vitro imaging assay revealed for the first time in a 3D microenvironment setting the statistically significant impact laser-microgrooved disk surfaces have on both cell adherence and recruitment of cells in proximity to the disk. It also allowed visualization of membrane protrusivity and cytoskeletal organization in cells adherent to the implant disk. Conclusion: This assay provides a simple and effective way of observing cell behavior on and around the implant disk surface in a more physiologic 3D setting. Within the limits of this study, it revealed that the laser-microgrooved implant surface demonstrates significant superiority in fibroblast recruitment and binding in a 3D microenvironment.   \n",
              "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Artificial intelligence (AI)-based image analysis is increasingly being used for preclinical safety-assessment studies in the pharmaceutical industry. In this paper, we present an AI-based solution for preclinical toxicology studies. We trained a set of algorithms to learn and quantify multiple typical histopathological findings in whole slide images (WSIs) of the livers of young Sprague Dawley rats by using a U-Net-based deep learning network. The trained algorithms were validated using 255 liver WSIs to detect, classify, and quantify seven types of histopathological findings (including vacuolation, bile duct hyperplasia, and single-cell necrosis) in the liver. The algorithms showed consistently good performance in detecting abnormal areas. Approximately 75% of all specimens could be classified as true positive or true negative. In general, findings with clear boundaries with the surrounding normal structures, such as vacuolation and single-cell necrosis, were accurately detected with high statistical scores. The results of quantitative analyses and classification of the diagnosis based on the threshold values between \"no findings\" and \"abnormal findings\" correlated well with diagnoses made by professional pathologists. However, the scores for findings ambiguous boundaries, such as hepatocellular hypertrophy, were poor. These results suggest that deep learning-based algorithms can detect, classify, and quantify multiple findings simultaneously on rat liver WSIs. Thus, it can be a useful supportive tool for a histopathological evaluation, especially for primary screening in rat toxicity studies. Copyright ?2022 The Japanese Society of Toxicologic Pathology.   \n",
              "\n",
              "        Primary Author  \\\n",
              "0          Janes, L.E.   \n",
              "1          Wahyono, T.   \n",
              "2          Yang, Na-Na   \n",
              "3  Kasherwal, Vishakha   \n",
              "4    Shimazaki, Taishi   \n",
              "\n",
              "                                                  Journal  Year  Volume Issue  \\\n",
              "0                   Journal of Hand Surgery Global Online  2022     4.0     1   \n",
              "1                                        Veterinary World  2022    15.0     2   \n",
              "2                                Natural Product Research  2021    35.0    24   \n",
              "3  International Journal of Oral & Maxillofacial Implants  2021    36.0     6   \n",
              "4                        Journal of toxicologic pathology  2022    35.0     2   \n",
              "\n",
              "       Pages  Comments   Eppi ID  target  \n",
              "0      32-39       NaN  14844675       0  \n",
              "1    331-340       NaN  14844676       0  \n",
              "2  6180-6183       NaN  14844677       0  \n",
              "3  1121-1128       NaN  14844678       0  \n",
              "4    135-147       NaN  14844679       0  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-e2048af0-caec-4856-966d-51b6fba75920\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Title</th>\n",
              "      <th>Abstract</th>\n",
              "      <th>Primary Author</th>\n",
              "      <th>Journal</th>\n",
              "      <th>Year</th>\n",
              "      <th>Volume</th>\n",
              "      <th>Issue</th>\n",
              "      <th>Pages</th>\n",
              "      <th>Comments</th>\n",
              "      <th>Eppi ID</th>\n",
              "      <th>target</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>An In Vivo Comparison: Novel Mesh Suture Versus Traditional Suture-Based Repair in a Rabbit Tendon Model</td>\n",
              "      <td>Purpose: Despite advancements in surgical techniques, suture pull-though and rupture continue to limit the early range of motion and functional rehabilitation after flexor tendon repairs. The aim of this study was to evaluate a suturable mesh compared with a commonly used braided suture in an in vivo rabbit intrasynovial tendon model. Method(s): Twenty-four New Zealand female rabbits (3-4 kg) were injected with 2 units/kg botulinum toxin evenly distributed into 4 sites in the left calf. After 1 week, the animals underwent surgical tenotomy of the flexor digitorum tendon and were randomized to repair with either 2-0 Duramesh suturable mesh or to 2-0 Fiberwire using a 2-strand modified Kessler and 6-0 polypropylene running epitendinous suture. Rabbits were killed at 2, 4, and 9 weeks after surgery. Result(s): Grouping across time points, 58.3% (7 of 12) of Duramesh repairs were found to be intact for the explant compared with 16.7% (2 of 12) of Fiberwire repairs (P = .09). At 2 weeks, the mean Duramesh repairs were significantly stronger than the Fiberwire repairs with a mean failure load of 50.7 +/- 12.7 N compared to 14.8 +/- 18.3 N (P = .02). The load supported by the Duramesh repairs at 2 weeks (mean 50.7 +/- 12.7 N) was similar to the load supported by both Fiberwire (52.2 +/- 13.6 N) and Duramesh (57.6 +/- 22.3 N) at 4 weeks. The strength of repair between Fiberwire and Duramesh at 4 weeks and 9 weeks was not significantly different. Conclusion(s): The 2-strand tendon repair with suturable mesh achieved significantly greater strength at 2 weeks than the conventional suture material. Future studies should evaluate the strength of repair prior to 2 weeks to determine the strength curve for this novel suture material. Clinical Relevance: This study evaluates the utility of a novel suturable mesh for flexor tendon repair in an in vivo rabbit model compared with conventional suture material.Copyright ? 2021 The Authors</td>\n",
              "      <td>Janes, L.E.</td>\n",
              "      <td>Journal of Hand Surgery Global Online</td>\n",
              "      <td>2022</td>\n",
              "      <td>4.0</td>\n",
              "      <td>1</td>\n",
              "      <td>32-39</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14844675</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Effects of urea supplementation on ruminal fermentation characteristics, nutrient intake, digestibility, and performance in sheep: A meta-analysis</td>\n",
              "      <td>Background and Aim: As a non-protein nitrogen source, urea is a popular, low cost, and easily obtained protein supplement. The objective of the present study was to perform a meta-analysis of the effects of urea supplementation on rumen fermentation and sheep performance. Material(s) and Method(s): A total of 32 experiments from 21 articles were compiled into a dataset. The levels of dietary urea varied from 0 to 31 g/kg of dry matter (DM). Parameters observed were rumen fermentation product, nutrient intake, nutrient digestibility, and sheep performance. This dataset was analyzed using a mixed model methodology, with urea supplementation levels as fixed effects and the different experiments as random effects. Result(s): Increasing levels of urea were associated with increases (p=0.008) in rumen pH, butyrate (C4) production, and ammonia (NH3-N) concentration. Urea supplementation had minor effects on total volatile fatty acids (p=0.242), total protozoa (p=0.429), and the microbial N supply (p=0.619), but tended to increase methane production (CH4; p&lt;0.001). Supplementation of urea increased the intake of dry matter (DM; p=0.004) and crude protein (CP; p=0.001). Digestibility parameters, such as DM digestibility (DMD) and CP digestibility (CPD), also increased (p&lt;0.01) as a result of urea supplementation. Retained N (p=0.042) and N intake (p&lt;0.001) were higher with increasing levels of urea supplementation. In terms of animal performance, supplementation of urea increased average daily gain (ADG; p=0.024), but decreased the hot carcass weight percentage (p=0.017). Conclusion(s): This meta-analysis reports the positive effects of urea supplementation on rumen fermentation products (i.e., pH, C4, and NH3-N), intake (DM, CP, and N), digestibility (DMD and CPD), and ADG in sheep.Copyright: Wahyono, et al. Open Access. This article is distributed under the terms of the Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/), which permits unrestricted use, distribution, and reproduction in any medium, provided you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons license, and indicate if changes were made. The Creative Commons Public Domain Dedication waiver (http://creativecommons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated.</td>\n",
              "      <td>Wahyono, T.</td>\n",
              "      <td>Veterinary World</td>\n",
              "      <td>2022</td>\n",
              "      <td>15.0</td>\n",
              "      <td>2</td>\n",
              "      <td>331-340</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14844676</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>The in vitro and in vivo anti-inflammatory activities of triterpene saponins from Clematis florida.</td>\n",
              "      <td>Clematis florida is widely used in She Ethnopharmacy in China owing to its significant anti-inflammatory activities. This study aimed to investigate the anti-inflammatory effect of the active fraction of C. florida (CFAF) in an arthritis animal model and its possible mechanism. Pre-inflammatory cytokine levels were examined by ELISA. CFAF can significantly improve the symptoms of arthritis such as paw swelling, arthritic index, and histological condition in AA rat. CFAF can also reduce levels of IL-1?, TNF-? and IL-6. Further studies showed that triterpene saponins from CFAF induced anti-inflammatory activity inhibited inflammatory mediators by blocking JAK/STAT signalling pathways in the LPS-treated macrophages.</td>\n",
              "      <td>Yang, Na-Na</td>\n",
              "      <td>Natural Product Research</td>\n",
              "      <td>2021</td>\n",
              "      <td>35.0</td>\n",
              "      <td>24</td>\n",
              "      <td>6180-6183</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14844677</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Evaluating Mouse Fibroblast Interaction with Implant Surfaces in a 3D Microenvironment.</td>\n",
              "      <td>Purpose: Previous studies assessing fibroblast interactions with implants have mainly relied on measurements such as cell migration, gene expression, and cell adhesion. For these studies, testing cellular behavior at the implant surface was done by imaging the cell-implant interface using standard microscopy techniques in 2D tissue culture dishes. The true behavior of cells relative to the implant can best be assessed in a more physiologic 3D microenvironment. Materials and Methods: The embedding of the implant disks in 3D collagen gels was standardized with labeled fibroblasts to allow the imaging of fibroblast morphology and behavior when proximal to or binding to the implant disks. This allowed comparison of the behavior of laser-microgrooved and machined implant disk surfaces quantitatively in an in vitro 3D microenvironment. Results: This in vitro imaging assay revealed for the first time in a 3D microenvironment setting the statistically significant impact laser-microgrooved disk surfaces have on both cell adherence and recruitment of cells in proximity to the disk. It also allowed visualization of membrane protrusivity and cytoskeletal organization in cells adherent to the implant disk. Conclusion: This assay provides a simple and effective way of observing cell behavior on and around the implant disk surface in a more physiologic 3D setting. Within the limits of this study, it revealed that the laser-microgrooved implant surface demonstrates significant superiority in fibroblast recruitment and binding in a 3D microenvironment.</td>\n",
              "      <td>Kasherwal, Vishakha</td>\n",
              "      <td>International Journal of Oral &amp; Maxillofacial Implants</td>\n",
              "      <td>2021</td>\n",
              "      <td>36.0</td>\n",
              "      <td>6</td>\n",
              "      <td>1121-1128</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14844678</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Deep learning-based image-analysis algorithm for classification and quantification of multiple histopathological lesions in rat liver.</td>\n",
              "      <td>Artificial intelligence (AI)-based image analysis is increasingly being used for preclinical safety-assessment studies in the pharmaceutical industry. In this paper, we present an AI-based solution for preclinical toxicology studies. We trained a set of algorithms to learn and quantify multiple typical histopathological findings in whole slide images (WSIs) of the livers of young Sprague Dawley rats by using a U-Net-based deep learning network. The trained algorithms were validated using 255 liver WSIs to detect, classify, and quantify seven types of histopathological findings (including vacuolation, bile duct hyperplasia, and single-cell necrosis) in the liver. The algorithms showed consistently good performance in detecting abnormal areas. Approximately 75% of all specimens could be classified as true positive or true negative. In general, findings with clear boundaries with the surrounding normal structures, such as vacuolation and single-cell necrosis, were accurately detected with high statistical scores. The results of quantitative analyses and classification of the diagnosis based on the threshold values between \"no findings\" and \"abnormal findings\" correlated well with diagnoses made by professional pathologists. However, the scores for findings ambiguous boundaries, such as hepatocellular hypertrophy, were poor. These results suggest that deep learning-based algorithms can detect, classify, and quantify multiple findings simultaneously on rat liver WSIs. Thus, it can be a useful supportive tool for a histopathological evaluation, especially for primary screening in rat toxicity studies. Copyright ?2022 The Japanese Society of Toxicologic Pathology.</td>\n",
              "      <td>Shimazaki, Taishi</td>\n",
              "      <td>Journal of toxicologic pathology</td>\n",
              "      <td>2022</td>\n",
              "      <td>35.0</td>\n",
              "      <td>2</td>\n",
              "      <td>135-147</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14844679</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e2048af0-caec-4856-966d-51b6fba75920')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e2048af0-caec-4856-966d-51b6fba75920 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e2048af0-caec-4856-966d-51b6fba75920');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-3e963622-4668-40ac-9223-c39f4e738f20\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-3e963622-4668-40ac-9223-c39f4e738f20')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-3e963622-4668-40ac-9223-c39f4e738f20 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "animal",
              "summary": "{\n  \"name\": \"animal\",\n  \"rows\": 2212,\n  \"fields\": [\n    {\n      \"column\": \"Title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2207,\n        \"samples\": [\n          \"Dissociating reward sensitivity and negative urgency effects on impulsivity in the five-choice serial reaction time task.\",\n          \"Gastrointestinal nematodes from buffalo in Minoufiya Governorate, Egypt with special reference to Bunostomum phlebotomum\",\n          \"Cinaciguat (BAY-582667) Modifies Cardiopulmonary and Systemic Circulation in Chronically Hypoxic and Pulmonary Hypertensive Neonatal Lambs in the Alto Andino.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2148,\n        \"samples\": [\n          \"Quality control for Chinese patent medicine (CPM) containing animal-derived crude drug(s) is rather difficult. The methods based on chemical composition analysis, which are commonly used in CPM consisted of plant-derived crude drugs, are often not applicable for CPM containing animal-derived crude drug, because the effective constituents of most animal-derived crude drugs remain unknown. Even if there are such methods, they are usually qualitative rather than quantitative, and the specificity is generally poor. Here we proposed a molecular quantification method for CPM containing animal-derived crude drug, based upon the hypothesis that the amount of remnant DNA fragments could reflect feeding quantity of the crude drugs and thus ensure the quality of the CPM. Take Jinlong capsule [a hepatocellular carcinoma-resisting Chinese patent medicine comprising of three fresh animal drugs, i.e. Shougong (Peking gecko, Gekko swinhonis), Qi She (sharp-snouted pitviper, Deinagkistrodon acutus), and Jinqian Baihua She (many-banded krait, Bungarus multicinctus)] as an example, we established a qPCR assay for Qi She in the capsule, which verified the feasibility of the quality control method based on molecular quantification. Species-specific primers and TaqMan probe for Qi She were designed, and the qPCR assay system was then established. The assay exhibited a good specificity; there's a good linearity between Ct values and logarithm of the target amplicon copy numbers within the range of 8.8 x 101 to 8.8 x 106 copies/muL, and the limit of detection was 88 copies/muL. The method was validated through reproducibility, stability assessment. Recovery of spiked samples was between 91.59% and 101.69%. It was verified that the copy numbers reflected the original feeding amount of an animal-derived crude drug by self-made Jinlong capsules. The assay was successfully applied in Qi She-specific amplicon determination in 20 batches of Jinlong capsule. The study was expected to provide a new strategy for quality control of CPM containing animal-derived crude drug.Copyright ? 2021 Elsevier B.V.\",\n          \"Background: Intrathecal ketamine has been studied extensively in animals, but rarely in humans. Intrathecal dexmedetomidine prolongs the duration of spinal anesthesia. Objective: To investigate the efficacy and safety of intrathecal dexmedetomidine, ketamine, or both when added to bupivacaine for postoperative analgesia in major abdominal cancer surgery. Design: Double?blinded, randomized, controlled trial. Setting: Academic medical center. Methods: Ninety patients were randomly allocated to receive either intrathecal 10 mg of hyperbaric bupivacaine 0.5% and 5 micro g of dexmedetomidine (group I, n = 30), 10 mg of hyperbaric bupivacaine 0.5% and 0.1 mg/kg ketamine (group II, n = 30), or 10 mg of hyperbaric bupivacaine 0.5% and 5 micro g of dexmedetomidine plus 0.1 mg/kg of ketamine (group III, n = 30). Hemodynamics, pain score, time to first request of analgesia, total PCA morphine consumption, sedation score, and adverse effects in the first 24 hours postoperatively were recorded. Results: Time to first request of analgesia was longer in group II (7.42 +/? 1.43 h) and group III (13.00 +/? 7.31h) compared to group I (3.50 +/? 1.57 h). PCA morphine consumption was less in group III (6.67 +/? 2.8 mg) compared to group I (9.16 +/? 3.63 mg) and group II (8.66 +/? 3.49 mg). Group III showed lower postoperative pain scores, and a higher incidence of postoperative sedation (P < 0.03). Limitations: This study is limited by its relatively small sample size. Conclusion: In conclusion, the combination of intrathecal dexmedetomidine and ketamine provided superior postoperative analgesia, prolonged the time to first request of rescue analgesia, and reduced the total consumption of PCA morphine, without serious side effects compared to either drug alone.\",\n          \"Deep brain stimulation has been successfully used in treatment of motor symptoms of Parkinson's disease and other movement disorders. In a recent multi?center prospectively randomized study, deep brain stimulation of the fornix was administered in order to ameliorate the cognitive symptoms and clinical course of Alzheimer's disease (AD). The study points to the possibility of modest slowing of the cognitive decline in AD in a subset of patients older than 65, while at the same time highlights the risk of stimulation in exacerbation of this decline in younger patients. The logic of conducting large clinical trials in the face of limited scientific understanding of the pathophysiology of AD and response of affected brain regions to electrical stimulation, is discussed with emphasis on the need to conduct: (i) animal studies in AD models, using precise focused stimulation; (ii) studies in patients who are implanted with depth electrodes for established clinical reasons (i.e., patients with epilepsy or movement disorders); and (iii) smaller adaptive studies in AD patients with systematic alterations of therapeutic parameters such as stimulation protocol. Copyright ? 2016 ? IOS Press and the authors. All rights reserved.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Primary Author\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2142,\n        \"samples\": [\n          \"Heidari, A.E.\",\n          \"Bors, Silviu-Ionut\",\n          \"Safwat, S.M.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1001,\n        \"samples\": [\n          \"Journal of veterinary internal medicine\",\n          \"Journal of developmental biology\",\n          \"Zoonoses and public health\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Year\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 2015,\n        \"max\": 2022,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          2021,\n          2015,\n          2022\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Volume\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2639.590298415877,\n        \"min\": 2.0,\n        \"max\": 122123.0,\n        \"num_unique_values\": 269,\n        \"samples\": [\n          31.0,\n          322.0,\n          134.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Issue\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 65,\n        \"samples\": [\n          \"6582\",\n          \"4gskglaxosmithkline\",\n          \"1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Pages\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1838,\n        \"samples\": [\n          \"iii137\",\n          \"889211\",\n          \"1095-1111\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Comments\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Eppi ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 638,\n        \"min\": 14844675,\n        \"max\": 14846886,\n        \"num_unique_values\": 2212,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"target\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 0,\n        \"num_unique_values\": 1,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 25
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "human.head()"
      ],
      "metadata": {
        "id": "757WbVrjp3z5",
        "outputId": "ab018caf-13ff-4745-d2f2-abc26e60a617",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                                                                                            Title  \\\n",
              "0                      How does pre-dialysis education need to change? Findings from a qualitative study with staff and patients.   \n",
              "1  Factors associated with regular eye examinations in people with diabetes: results from the Victorian Population Health Survey.   \n",
              "2                                        Alirocumab for the treatment of hyperlipidemia in high-risk patients: an updated review.   \n",
              "3                                Comparison of risk factors for hepatitis B and C in patients visiting a gastroenterology clinic.   \n",
              "4          Comparative safety of atorvastatin 80 mg versus 10 mg derived from analysis of 49 completed trials in 14,236 patients.   \n",
              "\n",
              "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Abstract  \\\n",
              "0  BACKGROUND: Pre-dialysis education (PDE) is provided to thousands of patients every year, helping them decide which renal replacement therapy (RRT) to choose. However, its effectiveness is largely unknown, with relatively little previous research into patients' views about PDE, and no research into staff views. This study reports findings relevant to PDE from a larger mixed methods study, providing insights into what staff and patients think needs to improve., METHODS: Semi-structured interviews in four hospitals with 96 clinical and managerial staff and 93 dialysis patients, exploring experiences of and views about PDE, and analysed using thematic framework analysis., RESULTS: Most patients found PDE helpful and staff valued its role in supporting patient decision-making. However, patients wanted to see teaching methods and materials improve and biases eliminated. Staff were less aware than patients of how informal staff-patient conversations can influence patients' treatment decision-making. Many staff felt ill equipped to talk about all treatment options in a balanced and unbiased way. Patient decision-making was found to be complex and patients' abilities to make treatment decisions were adversely affected in the pre-dialysis period by emotional distress., CONCLUSIONS: Suggested improvements to teaching methods and educational materials are in line with previous studies and current clinical guidelines. All staff, irrespective of their role, need to be trained about all treatment options so that informal conversations with patients are not biased. The study argues for a more individualised approach to PDE which is more like counselling than education and would demand a higher level of skill and training for specialist PDE staff. The study concludes that even if these improvements are made to PDE, not all patients will benefit, because some find decision-making in the pre-dialysis period too complex or are unable to engage with education due to illness or emotional distress. It is therefore recommended that pre-dialysis treatment decisions are temporary, and that PDE is replaced with on-going RRT education which provides opportunities for personalised education and on-going review of patients' treatment choices. Emotional support to help overcome the distress of the transition to end-stage renal disease will also be essential to ensure all patients can benefit from RRT education.   \n",
              "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               PURPOSE: Although diabetes increases the risk of becoming visually impaired or blind, a large proportion of people with diabetes are not receiving the recommended eye care to detect and prevent retinopathy. Assessing a broad range of demographic, health behavior, and societal characteristics in relation to eye care utilization, the present study aims to increase knowledge about the potential impact of such factors on eye care utilization., METHODS: In 2003, for the first time, the annual Victorian Population Health Survey (VPHS) incorporated various eye health-related questions. Approximately 12,600 primary approach letters were mailed to all eligible and randomly selected households. Using computer-assisted telephone interviewing (CATI), the interviewer selected the person aged 18 years or over with the most recent birthday within the contacted household., RESULTS: The mean age of all 7500 participants was 47.7 years (range, 18-99 years). Six percent (n = 424) of all participants had diabetes, of whom 80% (n = 345) reported a visit to an eye care specialist within the last 2 years. People with diabetes were more likely to have had an eye test within the last 2 years if they had seen a healthcare provider or had one of various health checks, including checks not related to diabetes, within the same time., CONCLUSION: Results suggest that people who take an interest in their general health may also be more aware of the importance of eye examinations to avoid vision loss. Eye health promotion activities therefore need to broaden their reach to approach from outside the health sector, targeting people with diabetes who normally do not receive health checks. The importance of dilated eye examinations for people with diabetes needs to be further promoted for eye care providers.   \n",
              "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 INTRODUCTION: Alirocumab is a fully human immunoglobulin G1 monoclonal antibody directed against proprotein convertase subtilisin/kexin type 9 (PCSK9) approved for the treatment of hypercholesterolemia in high-risk patients. The objective is to provide an updated review of the recent data published for alirocumab. Areas covered: The efficacy and safety of alirocumab has been initially evaluated in a comprehensive phase 3 program conducted in more than 6 000 patients with primary non-familial and heterozygous familial hypercholesterolemia: alirocumab reduced LDL-cholesterol up to 62% in phase 3 with every 2-week dosing compared with placebo, and up to 36% compared with ezetimibe, with an excellent safety and tolerability profile. Herein, the author describes new efficacy and safety data obtained from complementary analyses of the phase 3 program submitted for approval and reports data from new specific trials. Expert commentary: Based on current high pricing, the patient groups prioritized for alirocumab treatment are patients with heterozygous familial hypercholesterolemia and patients with atherosclerotic cardiovascular disease who have substantially elevated LDL cholesterol on maximally tolerated statin plus ezetimibe therapy. The ongoing ODYSSEY OUTCOMES trial will provide important information on the cost-effectiveness of alirocumab treatment.   \n",
              "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        OBJECTIVE: To find out and compare the risk factors for hepatitis B and C infections in patients visiting a gastroenterology clinic., DESIGN: A case-control study., PLACE AND DURATION OF STUDY: The Liver Stomach Clinic, Karachi, from July 2004 to September 2004., PATIENTS AND METHODS: Patients of hepatitis B and C visiting the clinic were interviewed and data were noted on a prescribed form. Patients with dyspeptic symptoms who were negative for both hepatitis B and C were taken as controls. Statistical analysis was done using SPSS package., RESULTS: Total numbers of patients interviewed were 148; 63 with hepatitis C, 41 with hepatitis B and 44 in the control group. These patients hailed from various parts of Pakistan with diverse ethnicity. Comparing hepatitis C with the control group, important risk factors identified were lower level of education, the occupational exposure to the blood and syringes, history of blood transfusions, taking therapeutic injections and intravenous drips, and habit of getting shaved by barbers. Patients of hepatitis B were younger as compared to the control group. Their knowledge about spread of infection was poor. These patients had not received hepatitis B vaccine during childhood. Less number of risk factors could be identified in this group, Shaving from the barber s shop was also found to be a risk factor just like in hepatitis C., CONCLUSION: There is a need to educate general population about the possible risk factors associated with the spread of hepatitis C and B. Proper screening of blood products and universal precautions against the spread of infections are recommended. Treatment by 1/V drips and getting shaved by barbers should be discouraged. Vaccination against hepatitis B is recommended.   \n",
              "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Atorvastatin has been shown to reduce coronary events and revascularization procedures in patients with multiple risk factors for coronary heart disease. Recent studies with atorvastatin 80 mg support the overall safety of this dose during long-term treatment. However, physicians appear reluctant to use high doses of statins. A retrospective analysis of pooled data from 49 clinical trials of atorvastatin in 14,236 patients treated for an average period of 2 weeks to 52 months was conducted. The study compared the safety of atorvastatin 10 mg (n = 7,258), atorvastatin 80 mg (n = 4,798), and placebo (n = 2,180) and included analyses on treatment-associated adverse events; nonserious and serious adverse events related to the musculoskeletal, hepatic, and renal systems; the incidence of elevations of creatine kinase >10 times the upper limit of normal (ULN); and hepatic transaminases >3 times ULN. Percentages of patients experiencing > or =1 adverse event were similar across all 3 groups. Withdrawals due to treatment-related adverse events were observed in 2.4%, 1.8%, and 1.2% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. Serious adverse events were rare and seldom led to treatment withdrawal with any dose. Treatment-associated myalgia was observed in 1.4%, 1.5%, and 0.7% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. No cases of rhabdomyolysis were reported in any group. Persistent elevations in hepatic transaminases >3 times ULN were observed in 0.1%, 0.6%, and 0.2% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. The incidence of treatment-associated adverse events for atorvastatin 80 mg was similar to that of atorvastatin 10 mg and placebo. In conclusion, the results of this analysis support the positive safety profile of atorvastatin at the highest dose.   \n",
              "\n",
              "    Primary Author  \\\n",
              "0     Combes, Gill   \n",
              "1  Muller, Andreas   \n",
              "2  Farnier, Michel   \n",
              "3     Shazi, Lubna   \n",
              "4   Newman, Connie   \n",
              "\n",
              "                                                                                    Journal  \\\n",
              "0                                                                            BMC nephrology   \n",
              "1  Optometry and vision science : official publication of the American Academy of Optometry   \n",
              "2                                                   Expert review of cardiovascular therapy   \n",
              "3                       Journal of the College of Physicians and Surgeons--Pakistan : JCPSP   \n",
              "4                                                        The American journal of cardiology   \n",
              "\n",
              "     Year Volume Issue    Pages  Comments   Eppi ID  target  \n",
              "0  2017.0     18     1      334       NaN  14842061       1  \n",
              "1  2006.0     83     2   96-101       NaN  14842062       1  \n",
              "2  2017.0     15    12  923-932       NaN  14842063       1  \n",
              "3  2006.0     16     2    104-7       NaN  14842064       1  \n",
              "4  2006.0     97     1     61-7       NaN  14842065       1  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4e8c82d7-bfce-4106-a483-832e21331acf\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Title</th>\n",
              "      <th>Abstract</th>\n",
              "      <th>Primary Author</th>\n",
              "      <th>Journal</th>\n",
              "      <th>Year</th>\n",
              "      <th>Volume</th>\n",
              "      <th>Issue</th>\n",
              "      <th>Pages</th>\n",
              "      <th>Comments</th>\n",
              "      <th>Eppi ID</th>\n",
              "      <th>target</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>How does pre-dialysis education need to change? Findings from a qualitative study with staff and patients.</td>\n",
              "      <td>BACKGROUND: Pre-dialysis education (PDE) is provided to thousands of patients every year, helping them decide which renal replacement therapy (RRT) to choose. However, its effectiveness is largely unknown, with relatively little previous research into patients' views about PDE, and no research into staff views. This study reports findings relevant to PDE from a larger mixed methods study, providing insights into what staff and patients think needs to improve., METHODS: Semi-structured interviews in four hospitals with 96 clinical and managerial staff and 93 dialysis patients, exploring experiences of and views about PDE, and analysed using thematic framework analysis., RESULTS: Most patients found PDE helpful and staff valued its role in supporting patient decision-making. However, patients wanted to see teaching methods and materials improve and biases eliminated. Staff were less aware than patients of how informal staff-patient conversations can influence patients' treatment decision-making. Many staff felt ill equipped to talk about all treatment options in a balanced and unbiased way. Patient decision-making was found to be complex and patients' abilities to make treatment decisions were adversely affected in the pre-dialysis period by emotional distress., CONCLUSIONS: Suggested improvements to teaching methods and educational materials are in line with previous studies and current clinical guidelines. All staff, irrespective of their role, need to be trained about all treatment options so that informal conversations with patients are not biased. The study argues for a more individualised approach to PDE which is more like counselling than education and would demand a higher level of skill and training for specialist PDE staff. The study concludes that even if these improvements are made to PDE, not all patients will benefit, because some find decision-making in the pre-dialysis period too complex or are unable to engage with education due to illness or emotional distress. It is therefore recommended that pre-dialysis treatment decisions are temporary, and that PDE is replaced with on-going RRT education which provides opportunities for personalised education and on-going review of patients' treatment choices. Emotional support to help overcome the distress of the transition to end-stage renal disease will also be essential to ensure all patients can benefit from RRT education.</td>\n",
              "      <td>Combes, Gill</td>\n",
              "      <td>BMC nephrology</td>\n",
              "      <td>2017.0</td>\n",
              "      <td>18</td>\n",
              "      <td>1</td>\n",
              "      <td>334</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14842061</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Factors associated with regular eye examinations in people with diabetes: results from the Victorian Population Health Survey.</td>\n",
              "      <td>PURPOSE: Although diabetes increases the risk of becoming visually impaired or blind, a large proportion of people with diabetes are not receiving the recommended eye care to detect and prevent retinopathy. Assessing a broad range of demographic, health behavior, and societal characteristics in relation to eye care utilization, the present study aims to increase knowledge about the potential impact of such factors on eye care utilization., METHODS: In 2003, for the first time, the annual Victorian Population Health Survey (VPHS) incorporated various eye health-related questions. Approximately 12,600 primary approach letters were mailed to all eligible and randomly selected households. Using computer-assisted telephone interviewing (CATI), the interviewer selected the person aged 18 years or over with the most recent birthday within the contacted household., RESULTS: The mean age of all 7500 participants was 47.7 years (range, 18-99 years). Six percent (n = 424) of all participants had diabetes, of whom 80% (n = 345) reported a visit to an eye care specialist within the last 2 years. People with diabetes were more likely to have had an eye test within the last 2 years if they had seen a healthcare provider or had one of various health checks, including checks not related to diabetes, within the same time., CONCLUSION: Results suggest that people who take an interest in their general health may also be more aware of the importance of eye examinations to avoid vision loss. Eye health promotion activities therefore need to broaden their reach to approach from outside the health sector, targeting people with diabetes who normally do not receive health checks. The importance of dilated eye examinations for people with diabetes needs to be further promoted for eye care providers.</td>\n",
              "      <td>Muller, Andreas</td>\n",
              "      <td>Optometry and vision science : official publication of the American Academy of Optometry</td>\n",
              "      <td>2006.0</td>\n",
              "      <td>83</td>\n",
              "      <td>2</td>\n",
              "      <td>96-101</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14842062</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Alirocumab for the treatment of hyperlipidemia in high-risk patients: an updated review.</td>\n",
              "      <td>INTRODUCTION: Alirocumab is a fully human immunoglobulin G1 monoclonal antibody directed against proprotein convertase subtilisin/kexin type 9 (PCSK9) approved for the treatment of hypercholesterolemia in high-risk patients. The objective is to provide an updated review of the recent data published for alirocumab. Areas covered: The efficacy and safety of alirocumab has been initially evaluated in a comprehensive phase 3 program conducted in more than 6 000 patients with primary non-familial and heterozygous familial hypercholesterolemia: alirocumab reduced LDL-cholesterol up to 62% in phase 3 with every 2-week dosing compared with placebo, and up to 36% compared with ezetimibe, with an excellent safety and tolerability profile. Herein, the author describes new efficacy and safety data obtained from complementary analyses of the phase 3 program submitted for approval and reports data from new specific trials. Expert commentary: Based on current high pricing, the patient groups prioritized for alirocumab treatment are patients with heterozygous familial hypercholesterolemia and patients with atherosclerotic cardiovascular disease who have substantially elevated LDL cholesterol on maximally tolerated statin plus ezetimibe therapy. The ongoing ODYSSEY OUTCOMES trial will provide important information on the cost-effectiveness of alirocumab treatment.</td>\n",
              "      <td>Farnier, Michel</td>\n",
              "      <td>Expert review of cardiovascular therapy</td>\n",
              "      <td>2017.0</td>\n",
              "      <td>15</td>\n",
              "      <td>12</td>\n",
              "      <td>923-932</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14842063</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Comparison of risk factors for hepatitis B and C in patients visiting a gastroenterology clinic.</td>\n",
              "      <td>OBJECTIVE: To find out and compare the risk factors for hepatitis B and C infections in patients visiting a gastroenterology clinic., DESIGN: A case-control study., PLACE AND DURATION OF STUDY: The Liver Stomach Clinic, Karachi, from July 2004 to September 2004., PATIENTS AND METHODS: Patients of hepatitis B and C visiting the clinic were interviewed and data were noted on a prescribed form. Patients with dyspeptic symptoms who were negative for both hepatitis B and C were taken as controls. Statistical analysis was done using SPSS package., RESULTS: Total numbers of patients interviewed were 148; 63 with hepatitis C, 41 with hepatitis B and 44 in the control group. These patients hailed from various parts of Pakistan with diverse ethnicity. Comparing hepatitis C with the control group, important risk factors identified were lower level of education, the occupational exposure to the blood and syringes, history of blood transfusions, taking therapeutic injections and intravenous drips, and habit of getting shaved by barbers. Patients of hepatitis B were younger as compared to the control group. Their knowledge about spread of infection was poor. These patients had not received hepatitis B vaccine during childhood. Less number of risk factors could be identified in this group, Shaving from the barber s shop was also found to be a risk factor just like in hepatitis C., CONCLUSION: There is a need to educate general population about the possible risk factors associated with the spread of hepatitis C and B. Proper screening of blood products and universal precautions against the spread of infections are recommended. Treatment by 1/V drips and getting shaved by barbers should be discouraged. Vaccination against hepatitis B is recommended.</td>\n",
              "      <td>Shazi, Lubna</td>\n",
              "      <td>Journal of the College of Physicians and Surgeons--Pakistan : JCPSP</td>\n",
              "      <td>2006.0</td>\n",
              "      <td>16</td>\n",
              "      <td>2</td>\n",
              "      <td>104-7</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14842064</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Comparative safety of atorvastatin 80 mg versus 10 mg derived from analysis of 49 completed trials in 14,236 patients.</td>\n",
              "      <td>Atorvastatin has been shown to reduce coronary events and revascularization procedures in patients with multiple risk factors for coronary heart disease. Recent studies with atorvastatin 80 mg support the overall safety of this dose during long-term treatment. However, physicians appear reluctant to use high doses of statins. A retrospective analysis of pooled data from 49 clinical trials of atorvastatin in 14,236 patients treated for an average period of 2 weeks to 52 months was conducted. The study compared the safety of atorvastatin 10 mg (n = 7,258), atorvastatin 80 mg (n = 4,798), and placebo (n = 2,180) and included analyses on treatment-associated adverse events; nonserious and serious adverse events related to the musculoskeletal, hepatic, and renal systems; the incidence of elevations of creatine kinase &gt;10 times the upper limit of normal (ULN); and hepatic transaminases &gt;3 times ULN. Percentages of patients experiencing &gt; or =1 adverse event were similar across all 3 groups. Withdrawals due to treatment-related adverse events were observed in 2.4%, 1.8%, and 1.2% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. Serious adverse events were rare and seldom led to treatment withdrawal with any dose. Treatment-associated myalgia was observed in 1.4%, 1.5%, and 0.7% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. No cases of rhabdomyolysis were reported in any group. Persistent elevations in hepatic transaminases &gt;3 times ULN were observed in 0.1%, 0.6%, and 0.2% of patients in the atorvastatin 10 mg, atorvastatin 80 mg, and placebo groups, respectively. The incidence of treatment-associated adverse events for atorvastatin 80 mg was similar to that of atorvastatin 10 mg and placebo. In conclusion, the results of this analysis support the positive safety profile of atorvastatin at the highest dose.</td>\n",
              "      <td>Newman, Connie</td>\n",
              "      <td>The American journal of cardiology</td>\n",
              "      <td>2006.0</td>\n",
              "      <td>97</td>\n",
              "      <td>1</td>\n",
              "      <td>61-7</td>\n",
              "      <td>NaN</td>\n",
              "      <td>14842065</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4e8c82d7-bfce-4106-a483-832e21331acf')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-4e8c82d7-bfce-4106-a483-832e21331acf button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-4e8c82d7-bfce-4106-a483-832e21331acf');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-27d8d8d1-8c28-4f77-9370-e8356aecf558\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-27d8d8d1-8c28-4f77-9370-e8356aecf558')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-27d8d8d1-8c28-4f77-9370-e8356aecf558 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "human",
              "summary": "{\n  \"name\": \"human\",\n  \"rows\": 2411,\n  \"fields\": [\n    {\n      \"column\": \"Title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2410,\n        \"samples\": [\n          \"Laparoscopic gastropexy for chronic gastric volvulus complicated by pathologic aerophagia in a boy.\",\n          \"Additional value of hybrid PET/CT fusion imaging vs. conventional CT scan alone in the staging and management of patients with malignant pleural mesothelioma.\",\n          \"Chemokine receptor (CCR2) genotype is associated with myocardial infarction and heart failure in patients under 65 years of age.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2001,\n        \"samples\": [\n          \"OBJECTIVE: Human pregnancy disorders such as preeclampsia are thought to involve variations in cytokine levels. It has been proposed that, in preeclamptic women, a balance favoring the Th1-type over the Th2-type cytokine profile determines local or systemic immunologic responses to pregnancy and that this may cause defective placental implantation and placental ischemia, which activate systemic endothelial cells. The purpose of this study was to determine whether cytokine expression differs in the maternal, choriodecidual, and fetal compartments, and between women with or without preeclampsia., METHODS: Plasma concentrations of interferon gamma (IFNgamma), interleukin (IL)-2, IL-4, and IL-10 were measured by enzyme-linked immunosorbent assay (ELISA) in samples obtained from maternal peripheral blood (MPB), choriodecidual (CD), and fetal cord (FC) blood compartments of 17 women with preeclampsia and in 15 normotensive women. Intracellular concentrations of IFNgamma and IL-2 in T lymphocytes were assessed by flow cytometry., RESULTS: Plasma IFNgamma concentrations in both MPB and CD compartments were significantly higher in preeclamptic than in normotensive women. Maternal plasma IL-4 concentration was significantly lower in preeclamptic than in normotensive women. Intracellular IFNgamma and IL-2 concentrations did not differ significantly between preeclamptic and normotensive women., CONCLUSIONS: The dominant Th1-type over Th2-type cytokine profile is evident in MPB, but not in the CD and FC blood compartments. This might reflect the complex cytokine networks in the fetal-placental interface and might involve trophoblasts or decidual and endothelial cells, which could account for the increased plasma IFNgamma concentration and T-helper cell number.\",\n          \"OBJECTIVE: To study the chromosomal content of spermatozoa that could be selected for intracytoplasmic sperm injection (ICSI) in cases of macrocephalic sperm head syndrome., DESIGN: Case report., SETTING: Obstetrics, gynecology, urology, and reproductive biology departments., PATIENT(S): Two infertile patient candidates for ICSI presenting with total teratozoospermia (100%) with mainly large-headed spermatozoa (91% and 82%, respectively)., INTERVENTION(S): Fluorescence in situ hybridization with X, Y, 18 centromeric probes on unselected spermatozoa (all migrated spermatozoa) and specifically on selected spermatozoa with normal-sized heads., MAIN OUTCOME MEASURE(S): Percentage of polyploid, diploid, aneuploid, and normal haploid spermatozoa, according to X, Y, 18 chromosome centromeric probes on selected spermatozoa (head size compatible with ICSI)., RESULT(S): All the nonselected spermatozoa were abnormal, diploid, or polyploid. The rate of normal ploidy (haploid cells) among the selected sperm population was 1 per 28 for patient 1 and 5 per 51 for patient 2., CONCLUSION(S): This very low proportion of normal haploid spermatozoa among selected spermatozoa contraindicated ICSI for the two patients. We suggest performance of this selection and analysis before including (or not), in an ICSI program, patients with macrocephalic sperm head syndrome, associated or not with preimplantation genetic diagnosis.\",\n          \"The interplay of genetic and environmental factors places first-degree relatives of individuals with premature coronary heart disease at greater risk of developing the disease than the general population. Disease processes, such as dyslipidemia, hypertension, and glucose and insulin metabolism, and lifestyle habits, such as eating and exercise patterns, as well as socioeconomic status aggregate in families with coronary heart disease. The degree of risk associated with a family history varies with the degree of relationship and the age at onset of disease. All individuals with a family history of premature heart disease should have a thorough coronary risk assessment performed, which can be initiated in an office visit. Absolute risk for coronary heart disease determination will predict the intensity of preventive interventions. This article reviews the components of risk determination and primary prevention in individuals with a strong family history of coronary heart disease.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Primary Author\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2346,\n        \"samples\": [\n          \"Friedl, Karl E\",\n          \"Wilder-Smith, Oliver H G\",\n          \"Afghani, Afrooz\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 805,\n        \"samples\": [\n          \"Cell and tissue research\",\n          \"Agri : Agri (Algoloji) Dernegi'nin Yayin organidir = The journal of the Turkish Society of Algology\",\n          \"The Journal of pediatrics\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Year\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 5.022233630218472,\n        \"min\": 1997.0,\n        \"max\": 2020.0,\n        \"num_unique_values\": 16,\n        \"samples\": [\n          2017.0,\n          2006.0,\n          2019.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Volume\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 232,\n        \"samples\": [\n          \"183\",\n          \"55\",\n          \"8\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Issue\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 51,\n        \"samples\": [\n          \"23\",\n          \"18\",\n          \"12s\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Pages\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2211,\n        \"samples\": [\n          \"637-8\",\n          \"674-8\",\n          \"253-4\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Comments\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Eppi ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 696,\n        \"min\": 14842061,\n        \"max\": 14844471,\n        \"num_unique_values\": 2411,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"target\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 1,\n        \"num_unique_values\": 1,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 39
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "animal.isnull().sum(), human.isnull().sum()\n"
      ],
      "metadata": {
        "id": "TV4NGOzZubv1",
        "outputId": "abf4a72a-d408-4888-d35b-dbeeb3559927",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "(Title                0\n",
              " Abstract            61\n",
              " Primary Author       4\n",
              " Journal              0\n",
              " Year                 0\n",
              " Volume              30\n",
              " Issue              687\n",
              " Pages              190\n",
              " Comments          2212\n",
              " Eppi ID              0\n",
              " target               0\n",
              " dtype: int64,\n",
              " Title                0\n",
              " Abstract           410\n",
              " Primary Author       0\n",
              " Journal              0\n",
              " Year                 2\n",
              " Volume              25\n",
              " Issue              113\n",
              " Pages                6\n",
              " Comments          2411\n",
              " Eppi ID              0\n",
              " target               0\n",
              " dtype: int64)"
            ]
          },
          "metadata": {},
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "animal.columns, human.columns"
      ],
      "metadata": {
        "id": "OStO2DSGsxcs",
        "outputId": "ae2a8e50-c60e-4df8-9d62-6a9a8088621c",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "(Index(['Title', 'Abstract', 'Primary Author', 'Journal', 'Year', 'Volume',\n",
              "        'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'],\n",
              "       dtype='object'),\n",
              " Index(['Title', 'Abstract', 'Primary Author', 'Journal', 'Year', 'Volume',\n",
              "        'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'],\n",
              "       dtype='object'))"
            ]
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Deleting unwanted columns"
      ],
      "metadata": {
        "id": "Q5nE3WZNrwjB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "human = human.drop(columns=['Primary Author', 'Journal', 'Year', 'Volume',\n",
        "       'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'], inplace=True)\n",
        "animal = animal.drop(columns=['Primary Author', 'Journal', 'Year', 'Volume',\n",
        "       'Issue', 'Pages', 'Comments', 'Eppi ID', 'target'], inplace=True)"
      ],
      "metadata": {
        "id": "6OtJxwrGrr6H"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "animal.head()"
      ],
      "metadata": {
        "id": "sXPwqbYshHo_",
        "outputId": "2aa76421-7ef0-4d63-a400-628555a30e11",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 141
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "error",
          "ename": "AttributeError",
          "evalue": "'NoneType' object has no attribute 'head'",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-44-4c546e539ac5>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0manimal\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'head'"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "animal_df = animal.iloc[:2212]"
      ],
      "metadata": {
        "id": "J6x46ZweiD3Y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "human_df = human.iloc[:2411]"
      ],
      "metadata": {
        "id": "OmI9YKOniDR-"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "vectorizer = TfidfVectorizer(max_features=1000)\n",
        "X_train_tfidf = vectorizer.fit_transform(TRAIN_PATH)\n",
        "X_test_tfidf = vectorizer.transform(TEST_PATH)"
      ],
      "metadata": {
        "id": "0kSC_RhKczrO",
        "outputId": "508d21d2-0387-49e6-84cf-17b0d3818dca",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 321
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "error",
          "ename": "ValueError",
          "evalue": "Iterable over raw text documents expected, string object received.",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-14-65c7a8751515>\u001b[0m in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mvectorizer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTfidfVectorizer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmax_features\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mX_train_tfidf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvectorizer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit_transform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mTRAIN_PATH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mX_test_tfidf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvectorizer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtransform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mTEST_PATH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/sklearn/feature_extraction/text.py\u001b[0m in \u001b[0;36mfit_transform\u001b[0;34m(self, raw_documents, y)\u001b[0m\n\u001b[1;32m   2131\u001b[0m             \u001b[0msublinear_tf\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msublinear_tf\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2132\u001b[0m         )\n\u001b[0;32m-> 2133\u001b[0;31m         \u001b[0mX\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit_transform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mraw_documents\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2134\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_tfidf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2135\u001b[0m         \u001b[0;31m# X is already a transformed view of raw_documents so\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/sklearn/feature_extraction/text.py\u001b[0m in \u001b[0;36mfit_transform\u001b[0;34m(self, raw_documents, y)\u001b[0m\n\u001b[1;32m   1363\u001b[0m         \u001b[0;31m# TfidfVectorizer.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1364\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mraw_documents\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1365\u001b[0;31m             raise ValueError(\n\u001b[0m\u001b[1;32m   1366\u001b[0m                 \u001b[0;34m\"Iterable over raw text documents expected, string object received.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1367\u001b[0m             )\n",
            "\u001b[0;31mValueError\u001b[0m: Iterable over raw text documents expected, string object received."
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "5MswC83CeS4z"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}